### ISSUE TITLE
Fix S4049 FP: Do not raise on methods with generic parameters

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#8739]
As seen on [SonarCloud](https://sonarcloud.io/project/issues?directories=analyzers%2Ftests%2FSonarAnalyzer.TestFramework%2FBuild&resolutions=WONTFIX&id=sonaranalyzer-dotnet&open=AY1BVhSq1y3KG8AwrxZm)

The rule raises here and that cannot be turned into a property due to the generic parameter of the method:
```
    public IEnumerable<TSyntaxNodeType> GetNodes<TSyntaxNodeType>() where TSyntaxNodeType : SyntaxNode =>
        SyntaxTree.GetRoot().DescendantNodes().OfType<TSyntaxNodeType>();
```
